be/src/vec/columns/column_json.h

```diff
@@ -0,0 +1,299 @@
+
+        chars.resize(new_size);
+        memcpy(chars.data() + old_size, s.get_value(), size_to_append);
+        offsets.push_back(new_size);
+    }
+
+#if !__clang__
+#pragma GCC diagnostic pop
+#endif
+
+    void insert_from(const IColumn& src_, size_t n) override {
+        const ColumnJson& src = assert_cast<const ColumnJson&>(src_);
+        const size_t size_to_append =
+                src.offsets[n] - src.offsets[n - 1]; /// -1th index is Ok, see PaddedPODArray.
+
+        if (size_to_append == 1) {
+            /// shortcut for empty json
+            chars.push_back(0);
+            offsets.push_back(chars.size());
+        } else {
+            const size_t old_size = chars.size();
+            const size_t offset = src.offsets[n - 1];
+            const size_t new_size = old_size + size_to_append;
+
+            chars.resize(new_size);
+            memcpy_small_allow_read_write_overflow15(chars.data() + old_size, &src.chars[offset],
+                                                     size_to_append);
+            offsets.push_back(new_size);
+        }
+    }
+
+    void insert_data(const char* pos, size_t length) override {
+        const size_t old_size = chars.size();
+        const size_t new_size = old_size + length + 1;
+
+        chars.resize(new_size);
+        if (length) memcpy(chars.data() + old_size, pos, length);
+        chars[old_size + length] = 0;
+        offsets.push_back(new_size);
+    }
+
+    void insert_many_binary_data(char* data_array, uint32_t* len_array,
+                                 uint32_t* start_offset_array, size_t num) override {
+        for (size_t i = 0; i < num; i++) {
+            uint32_t len = len_array[i];
+            uint32_t start_offset = start_offset_array[i];
+            insert_data(data_array + start_offset, len);
+        }
+    };
+
+    void insert_many_dict_data(const int32_t* data_array, size_t start_index, const StringRef* dict,
+                               size_t num, uint32_t /*dict_num*/) {
+        for (size_t end_index = start_index + num; start_index < end_index; ++start_index) {
+            int32_t codeword = data_array[start_index];
+            insert_data(dict[codeword].data, dict[codeword].size);
+        }
+    }
+
+    void pop_back(size_t n) override {
+        size_t nested_n = offsets.back() - offset_at(offsets.size() - n);
+        chars.resize(chars.size() - nested_n);
+        offsets.resize_assume_reserved(offsets.size() - n);
+    }
+
+    StringRef serialize_value_into_arena(size_t n, Arena& arena, char const*& begin) const override;
+
+    const char* deserialize_and_insert_from_arena(const char* pos) override;
+
+    void update_hash_with_value(size_t n, SipHash& hash) const override {
+        size_t string_size = size_at(n);
+        size_t offset = offset_at(n);
+
+        hash.update(reinterpret_cast<const char*>(&string_size), sizeof(string_size));
+        hash.update(reinterpret_cast<const char*>(&chars[offset]), string_size);
+    }
+
```

Comment:
get_max_row_byte_size, serialize_vec, serialize_vec_with_null_map not implemented as ColumnString